soc/arm/st_stm32/common/soc_config.c

```diff
@@ -30,6 +32,34 @@ static int st_stm32_common_config(const struct device *dev)
 	DBGMCU->CR |= DBGMCU_CR_TRACE_IOEN;
 #endif
 
+#if defined(CONFIG_USE_SEGGER_RTT)
+#if defined(__HAL_RCC_DMA1_CLK_ENABLE)
+	/* On some STM32 boards, for unclear reason,
+	 * RTT feature is working with realtime update only when
+	 *   - one of the DMA is clocked.
+	 * See https://github.com/zephyrproject-rtos/zephyr/issues/34324
+	 */
```

Comment:
Put the comment before `#if defined(__HAL_RCC_DMA1_CLK_ENABLE)`

Reply:
Done